### ISSUE TITLE
Prepending component attributes causes an exception when said attribute is missing in calling template

### ISSUE THREAD

**Issue by panda-madness** [laravel/framework#35128]
- Laravel Version: 8.13
- PHP Version: 7.4
- Database Driver & Version: none

### Description:

Rendering a component with prepended attributes but without overriding that attribute in the calling template yields an error.

### Steps To Reproduce:

- `composer create-project laravel/laravel testing`
- `php artisan make:component TestComponent`
- Copy the example from the docs into the template

```blade
<div {{ $attributes->merge(['data-controller' => $attributes->prepends('profile-controller')]) }}>
    {{ $slot }}
</div>
```
- attempt to render the component without 'data-controller' specified.

Rendering this component like so: `<x-test-component data-controller="some-controller" />` works as expected.

Rendering this component like so: `<x-test-component />` yields the following error:
<img width="1209" alt="Screen Shot 2020-11-06 at 13 59 09" src="https://user-images.githubusercontent.com/6180087/98340888-41417b80-2038-11eb-852c-a6815f5f7b72.png">

I would expect to have the string `profile-controller` to be the default and only value present if no attribute named `data-controller` is specified.

**Comment by panda-madness** [comment]
I've hotfixed this locally by adding a `__toString` method to [AppendableAttributeValue](https://github.com/laravel/framework/blob/8.x/src/Illuminate/View/AppendableAttributeValue.php) like so:

```
<?php

namespace Illuminate\View;

class AppendableAttributeValue
{
    /**
     * The attribute value.
     *
     * @var mixed
     */
    public $value;

    /**
     * Create a new appendable attribute value.
     *
     * @param  mixed  $value
     * @return void
     */
    public function __construct($value)
    {
        $this->value = $value;
    }

    public function __toString()
    {
        return (string)$this->value;
    }
}

```

**Comment by driesvints** [comment]
> Copy the example from the docs into the template

Which example?

**Comment by panda-madness** [comment]
@driesvints the one from the [Non-class attribute merging](https://laravel.com/docs/8.x/blade#non-class-attribute-merging) section. I've provided it verbatim in my original post.

**Comment by driesvints** [comment]
Copy paste your `TestComponent` please.

**Comment by panda-madness** [comment]
It's just an empty default Component stub:
```
<?php

namespace App\View\Components;

use Illuminate\View\Component;

class TestComponent extends Component
{
    /**
     * Create a new component instance.
     *
     * @return void
     */
    public function __construct()
    {
        //
    }

    /**
     * Get the view / contents that represent the component.
     *
     * @return \Illuminate\Contracts\View\View|string
     */
    public function render()
    {
        return view('components.test-component');
    }
}

```

**Comment by driesvints** [comment]
Thanks. I managed to reproduce this. I think the `__toString` solution is a valid one. Can you PR it? A test would be useful as well.

**Comment by panda-madness** [comment]
Sure, I'll link it here when it's ready.

**Comment by panda-madness** [comment]
Fixed in #35131
